### PR TITLE
Auth inter mgmt impl

### PR DESCRIPTION
**Pull request by MaGaMeGa**

### CODE REVIEW

File: authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java [hunk 1]
```diff
@@ -41,4 +109,351 @@
...
+			if (page == null || size == null) {
+				throw new BadPayloadException("Defined page or size could not be with undefined size or page.", HttpStatus.SC_BAD_REQUEST, CommonConstants.AUTHORIZATIOIN_URI + INTER_CLOUD_AUTHORIZATION_MGMT_URI);
+			} else {
+				validatedPage = page;
+				validatedSize = size;
+			}
+		}
+		final Direction validatedDirection = calculateDirection(direction, CommonConstants.AUTHORIZATIOIN_URI + INTER_CLOUD_AUTHORIZATION_MGMT_URI);
+		
+		final InterCloudAuthorizationListResponseDTO intraCloudAuthorizationEntriesResponse = authorizationDBService.getInterCloudAuthorizationEntriesResponse(validatedPage, validatedSize, validatedDirection, sortField);
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java]: rename variable to inter*

File: authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java [hunk 2]
```diff
@@ -41,4 +109,351 @@
...
+	@PostMapping(path = INTER_CLOUD_AUTHORIZATION_MGMT_URI, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+	@ResponseStatus(value = org.springframework.http.HttpStatus.CREATED)
+	@ResponseBody public InterCloudAuthorizationListResponseDTO addInterCloudAuthorization(@RequestBody final InterCloudAuthorizationRequestDTO request) {
+		logger.debug("New InterCloudAuthorization registration request recieved");
+		
+		final boolean isCloudIdNotValid = request.getCloudId() == null || request.getCloudId() < 1  ;
+		final boolean isServiceDefinitionNotValid = request.getServiceDefinitionIdList() == null || request.getServiceDefinitionIdList().isEmpty() ;
+		if (isCloudIdNotValid || isServiceDefinitionNotValid) {
+			String exceptionMsg = isCloudIdNotValid ? "Cloud Id is not valid" : "";
+			exceptionMsg = isServiceDefinitionNotValid ? exceptionMsg + " ServiceDefinition is null" :  exceptionMsg ;
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java]: null or blank 
use commas between errors

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 3]
```diff
@@ -29,4 +58,403 @@
...
+			throw ex;
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+	public InterCloudAuthorizationListResponseDTO createInterCloudAuthorizationResponse(Long cloudId,
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: cloudId is primitive long

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 4]
```diff
@@ -29,4 +58,403 @@
...
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+		
+				
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)	
+	public Page<InterCloudAuthorization> createInterCloudAuthorization(Long cloudId, List<Long> serviceDefinitionIds) {
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: cloudId is primitive long

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 5]
```diff
@@ -29,4 +58,403 @@
...
+		
+		if (cloudId < 1) {
+			throw new InvalidParameterException("Cloud id can't be null and must be greater than 0.");
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: remove null from error message

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 6]
```diff
@@ -29,4 +58,403 @@
...
+		}
+		for (final Long id : serviceDefinitionIds) {
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: check if the serviceDefinitionsIds list is null

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 7]
```diff
@@ -29,4 +58,403 @@
...
+		} catch (final InvalidParameterException ex) {
+			throw ex;
+		} catch (final Exception ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw new ArrowheadException(CommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
+		}
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	private InterCloudAuthorization createNewInterCloudAuthorization(final Long cloudId, final Long serviceDefinitionId) {
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: serviceDefinitionId is primitive long
use Cloud entity instead of cloudId

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 8]
```diff
@@ -29,4 +58,403 @@
...
+		final boolean cloudIdIsInvalid = cloudId < 1;
+		final boolean serviceDefinitionIdIsInvalid = serviceDefinitionId < 1;
+		
+		if ( cloudIdIsInvalid || serviceDefinitionIdIsInvalid) {
+			String exceptionMessage = "Following id parameters are invalid: ";
+			exceptionMessage = cloudIdIsInvalid ? exceptionMessage : exceptionMessage + "cloudId" ;
+			exceptionMessage = serviceDefinitionIdIsInvalid ? exceptionMessage : exceptionMessage + " serviceDefinitionId";
+			throw new InvalidParameterException(exceptionMessage);
+		}	
+
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java:540]: remove check

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 9]
```diff
@@ -29,4 +58,403 @@
...
+	//-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)	
+	public Page<InterCloudAuthorization> createInterCloudAuthorization(Long cloudId, List<Long> serviceDefinitionIds) {
+		
+		if (cloudId < 1) {
+			throw new InvalidParameterException("Cloud id can't be null and must be greater than 0.");
+		}
+		for (final Long id : serviceDefinitionIds) {
+			if (id == null || id < 1) {
+				throw new InvalidParameterException("SerdviceDefinition id can't be null and must be greater than 0.");
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: typo service

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 10]
```diff
@@ -29,4 +58,403 @@
...
+			
+			for (final Long serviceId : serviceDefinitionIds) {
+				entriesToSave.add(createNewInterCloudAuthorization(cloudId, serviceId));
+			}
+				
+			return new PageImpl<InterCloudAuthorization>(interCloudAuthorizationRepository.saveAll(entriesToSave));
+			
+		} catch (final InvalidParameterException ex) {
+			logger.debug(ex.getMessage(), ex);
+			throw ex;
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java:365]: remove throw ex

File: authorization/src/test/java/eu/arrowhead/core/authorization/AuthorizationControllerInterCloudTest.java [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	// Test of registerInterCloudAuthorization
+	
+	@Test
+	public void testRegisterInterCloudAuthorizationWithInvalidCloudId() throws Exception {
+		this.mockMvc.perform(post(INTER_CLOUD_AUTHORIZATION_MGMT_URI)
+			.contentType(MediaType.APPLICATION_JSON)
+			.content(objectMapper.writeValueAsBytes(new InterCloudAuthorizationRequestDTO((long) 0, createIdList(1, 2))))
```
Comment (rbocsi) [inline authorization/src/test/java/eu/arrowhead/core/authorization/AuthorizationControllerInterCloudTest.java]: use 0L

File: authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	//Tests of createInterCloudAuthorization
+	
+	@Test (expected = InvalidParameterException.class)
+	public void testCreateInterCloudAuthorizationResponseWithInvalidCloudId() {
+		authorizationDBService.createInterCloudAuthorization(-1L, List.of(1L));		
+	}
+
```
Comment (rbocsi) [inline authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java:98]: missing test to check serviceDefIds list/set is null

File: authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+	@Test (expected = InvalidParameterException.class)
+	public void testCreateInterCloudAuthorizationResponseWithNotExistingServiceDefintition() {
+		when(cloudRepository.findById(anyLong())).thenReturn(Optional.of(new Cloud()));
+		when(serviceDefinitionRepository.findById(anyLong())).thenReturn(Optional.ofNullable(null));
+		when(interCloudAuthorizationRepository.findByCloudIdAndServiceDefinitionId(anyLong(), anyLong())).thenReturn(Optional.ofNullable(null));
+		authorizationDBService.createInterCloudAuthorization(1L, List.of(Long.MAX_VALUE));
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Test (expected = InvalidParameterException.class)
```
Comment (rbocsi) [inline authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java]: remove this test

File: core-common/src/main/java/eu/arrowhead/common/CommonConstants.java [hunk 14]
```diff
@@ -49,10 +61,13 @@
 	
 	public static final String SERVER_ERROR_URI = "/error";
 	public static final String MGMT_URI = "/mgmt";
+	public static final String AUTHORIZATIOIN_URI = "/authorization";
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/CommonConstants.java]: remove this

File: core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java [hunk 15]
```diff
@@ -1,10 +1,15 @@
...
+
 import org.springframework.stereotype.Repository;
 
 import eu.arrowhead.common.database.entity.InterCloudAuthorization;
+import eu.arrowhead.common.database.entity.IntraCloudAuthorization;
 
 @Repository
 public interface InterCloudAuthorizationRepository extends RefreshableRepository<InterCloudAuthorization,Long> {
 
+	Optional<IntraCloudAuthorization> findByCloudIdAndServiceDefinitionId(long cloudId, long serviceDefinitionId);
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java]: change this to use entities

File: core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+	private String updatedAt;
+	
+	//=================================================================================================
+	// methods
+	
+	//-------------------------------------------------------------------------------------------------
+	public CloudResponseDTO() {}
+	
+	//-------------------------------------------------------------------------------------------------
+	public CloudResponseDTO(long id, String operator, String name, String address, int port,
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java]: finals

File: core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+	
+	private static final long serialVersionUID = 5183291727291459583L;
+	
+	private long id;
+	private String operator;
+	private String name;
+	private String address;
+	private int port;
+	private String gateKeeperServiceUri;
+	private Boolean secure;
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java]: use primitive boolean 3x

File: core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationRequestDTO.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+	private List<Long> serviceDefinitionIdList;
+	
+	//=================================================================================================
+	// methods
+	
+	//-------------------------------------------------------------------------------------------------
+	public InterCloudAuthorizationRequestDTO() {}
+	
+	//-------------------------------------------------------------------------------------------------	
+	public InterCloudAuthorizationRequestDTO(long cloudId,
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationRequestDTO.java]: use Long

File: core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationResponseDTO.java [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+	private String updatedAt;
+	
+	//=================================================================================================
+	// methods
+	
+	//-------------------------------------------------------------------------------------------------
+	public InterCloudAuthorizationResponseDTO() {}
+	
+	//-------------------------------------------------------------------------------------------------	
+	public InterCloudAuthorizationResponseDTO(long id, CloudResponseDTO cloud,
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationResponseDTO.java]: finals

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java (hunk 1 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java (hunk 2 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/AuthorizationController.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 3 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 4 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 5 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 6 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 9 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 10 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java:365]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 7 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 8 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java:540]: done

File: authorization/src/test/java/eu/arrowhead/core/authorization/AuthorizationControllerInterCloudTest.java (hunk 11 above)
Comment (MaGaMeGa) [reply authorization/src/test/java/eu/arrowhead/core/authorization/AuthorizationControllerInterCloudTest.java]: done

File: authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java (hunk 13 above)
Comment (MaGaMeGa) [reply authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java]: done

File: core-common/src/main/java/eu/arrowhead/common/CommonConstants.java (hunk 14 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/CommonConstants.java]: done

File: core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java (hunk 15 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java]: done

File: core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java (hunk 17 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java]: done

File: core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java (hunk 16 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/dto/CloudResponseDTO.java]: done

File: core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationRequestDTO.java (hunk 18 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationRequestDTO.java]: done

File: core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationResponseDTO.java (hunk 19 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/dto/InterCloudAuthorizationResponseDTO.java]: done

File: authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java (hunk 12 above)
Comment (MaGaMeGa) [reply authorization/src/test/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBServiceInterCloudTest.java:98]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 20]
```diff
@@ -410,50 +425,54 @@ private void checkConstraintsOfIntraCloudAuthorizationTable(final long consumerI
 	}
 	
 	//-------------------------------------------------------------------------------------------------
-	private void checkConstraintsOfInterCloudAuthorizationTable(final long cloudId, final long serviceDefinitionId) {
-		logger.debug("checkConstraintsOfInterCloudAuthorizationTable started..");
+	private void checkConstraintsOfInterCloudAuthorizationTable(final Cloud cloud, final ServiceDefinition serviceDefinition) {
+		logger.debug("checkConstraintsOfInterCloudAuthorizationTable started...");
 		
 		try {
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: remove try/catch

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 21]
```diff
@@ -410,50 +425,54 @@ private void checkConstraintsOfIntraCloudAuthorizationTable(final long consumerI
...
-		logger.debug("createInterCloudAuthorizationLis started..");
+	private InterCloudAuthorization createNewInterCloudAuthorization(final Cloud cloud, final long serviceDefinitionId) {
+		logger.debug("createInterCloudAuthorizationLis started...");
 		
-		final boolean cloudIdIsInvalid = cloudId < 1;
 		final boolean serviceDefinitionIdIsInvalid = serviceDefinitionId < 1;
 		
-		if ( cloudIdIsInvalid || serviceDefinitionIdIsInvalid) {
+		if (serviceDefinitionIdIsInvalid) {
 			String exceptionMessage = "Following id parameters are invalid: ";
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: use one line instead of 3

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 22]
```diff
@@ -410,50 +425,54 @@ private void checkConstraintsOfIntraCloudAuthorizationTable(final long consumerI
...
 		
-		final Optional<Cloud> cloud = cloudRepository.findById(cloudId);
-		final Optional<ServiceDefinition> serviceDefinition = serviceDefinitionRepository.findById(serviceDefinitionId);
-		if (cloud.isEmpty() || serviceDefinition.isEmpty()) {
-			String exceptionMessage = "Following entities are not availables: ";
-			exceptionMessage = cloud.isEmpty() ? exceptionMessage + "consumer with id: " + cloudId : exceptionMessage;
-			exceptionMessage = serviceDefinition.isEmpty() ? exceptionMessage + " serviceDefinition with id: " + serviceDefinitionId : exceptionMessage;
+		final Optional<ServiceDefinition> serviceDefinitionOptional = serviceDefinitionRepository.findById(serviceDefinitionId);
+		if (serviceDefinitionOptional.isEmpty()) {
+			String exceptionMessage = "Following id parameters are not present in database: ";
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: use one line

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 23]
```diff
@@ -410,50 +425,54 @@ private void checkConstraintsOfIntraCloudAuthorizationTable(final long consumerI
...
+		}else {
+			final ServiceDefinition serviceDefinition = serviceDefinitionOptional.get();
+			
+			try {
+				checkConstraintsOfInterCloudAuthorizationTable(cloud, serviceDefinition);
+				final InterCloudAuthorization interCloudAuthorization = new InterCloudAuthorization(cloud, serviceDefinition);				
+				return interCloudAuthorization;
+				
+			}catch(final InvalidParameterException ex) {
+				logger.info(ex.getMessage());
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: debug

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java [hunk 24]
```diff
@@ -410,50 +425,54 @@ private void checkConstraintsOfIntraCloudAuthorizationTable(final long consumerI
...
+		if (serviceDefinitionOptional.isEmpty()) {
+			String exceptionMessage = "Following id parameters are not present in database: ";
+			exceptionMessage =  exceptionMessage + " serviceDefinitionId :" + serviceDefinitionId;
 			throw new InvalidParameterException(exceptionMessage);
+		}else {
+			final ServiceDefinition serviceDefinition = serviceDefinitionOptional.get();
+			
+			try {
+				checkConstraintsOfInterCloudAuthorizationTable(cloud, serviceDefinition);
+				final InterCloudAuthorization interCloudAuthorization = new InterCloudAuthorization(cloud, serviceDefinition);
```
Comment (rbocsi) [inline authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: return new ...

File: core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java [hunk 25]
```diff
@@ -4,12 +4,14 @@
...
+import eu.arrowhead.common.database.entity.Cloud;
 import eu.arrowhead.common.database.entity.InterCloudAuthorization;
 import eu.arrowhead.common.database.entity.IntraCloudAuthorization;
+import eu.arrowhead.common.database.entity.ServiceDefinition;
 
 @Repository
 public interface InterCloudAuthorizationRepository extends RefreshableRepository<InterCloudAuthorization,Long> {
 
-	Optional<IntraCloudAuthorization> findByCloudIdAndServiceDefinitionId(long cloudId, long serviceDefinitionId);
+	Optional<IntraCloudAuthorization> findByCloudAndServiceDefinition(Cloud cloud, ServiceDefinition serviceDefinition);
```
Comment (rbocsi) [inline core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java]: finals

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 20 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 21 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 22 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 24 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java (hunk 23 above)
Comment (MaGaMeGa) [reply authorization/src/main/java/eu/arrowhead/core/authorization/database/service/AuthorizationDBService.java]: done

File: core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java (hunk 25 above)
Comment (MaGaMeGa) [reply core-common/src/main/java/eu/arrowhead/common/database/repository/InterCloudAuthorizationRepository.java]: done

**Review by rbocsi — COMMENTED** [review]
seems ok

**Review by rbocsi — APPROVED** [review]
